diff -up vault/templates/_helpers.tpl.0.26.1 vault/templates/_helpers.tpl
--- vault/templates/_helpers.tpl.0.26.1	2023-11-07 14:06:52.285821136 +0100
+++ vault/templates/_helpers.tpl	2023-11-07 14:07:45.445038627 +0100
@@ -738,6 +738,35 @@ Sets extra vault server Service annotati
 {{- end -}}
 
 {{/*
+Sets extra vault server Service internal annotations
+*/}}
+{{- define "vault.service.internal.annotations" -}}
+  {{- if .Values.server.service.internal.annotations }}
+    {{- $tp := typeOf .Values.server.service.internal.annotations }}
+    {{- if eq $tp "string" }}
+      {{- tpl .Values.server.service.internal.annotations . | nindent 4 }}
+    {{- else }}
+      {{- toYaml .Values.server.service.internal.annotations | nindent 4 }}
+    {{- end }}
+  {{- end }}
+{{- end -}}
+{{/*
+Sets extra vault server Service nonha annotations
+Note: We call it 'nonha' as we need to differentiate the "vault.service.annotations" which are
+      applied to all services
+*/}}
+{{- define "vault.service.nonha.annotations" -}}
+  {{- if .Values.server.service.nonha.annotations }}
+    {{- $tp := typeOf .Values.server.service.nonha.annotations }}
+    {{- if eq $tp "string" }}
+      {{- tpl .Values.server.service.nonha.annotations . | nindent 4 }}
+    {{- else }}
+      {{- toYaml .Values.server.service.nonha.annotations | nindent 4 }}
+    {{- end }}
+  {{- end }}
+{{- end -}}
+
+{{/*
 Sets PodSecurityPolicy annotations
 */}}
 {{- define "vault.psp.annotations" -}}
diff -up vault/templates/server-headless-service.yaml.0.26.1 vault/templates/server-headless-service.yaml
--- vault/templates/server-headless-service.yaml.0.26.1	2023-11-07 14:08:24.302197609 +0100
+++ vault/templates/server-headless-service.yaml	2023-11-07 14:08:48.707297472 +0100
@@ -21,6 +21,7 @@ metadata:
     vault-internal: "true"
   annotations:
 {{ template "vault.service.annotations" .}}
+{{ template "vault.service.internal.annotations" .}}
 spec:
   {{- if (semverCompare ">= 1.23-0" .Capabilities.KubeVersion.Version) }}
   {{- if .Values.server.service.ipFamilyPolicy }}
diff -up vault/templates/server-service.yaml.0.26.1 vault/templates/server-service.yaml
--- vault/templates/server-service.yaml.0.26.1	2023-11-07 14:09:43.152520231 +0100
+++ vault/templates/server-service.yaml	2023-11-07 14:09:55.406570360 +0100
@@ -20,6 +20,7 @@ metadata:
     app.kubernetes.io/managed-by: {{ .Release.Service }}
   annotations:
 {{ template "vault.service.annotations" .}}
+{{ template "vault.service.nonha.annotations" .}}
 spec:
   {{- if .Values.server.service.type}}
   type: {{ .Values.server.service.type }}
diff -up vault/values.schema.json.0.26.1 vault/values.schema.json
--- vault/values.schema.json.0.26.1	2023-11-07 14:10:35.177733085 +0100
+++ vault/values.schema.json	2023-11-07 14:11:52.244048399 +0100
@@ -931,6 +931,28 @@
                                 }
                             }
                         },
+                        "internal": {
+                            "type": "object",
+                            "properties": {
+                                "annotations": {
+                                    "type": [
+                                        "object",
+                                        "string"
+                                    ]
+                                }
+                            }
+                        },
+                        "nonha": {
+                            "type": "object",
+                            "properties": {
+                                "annotations": {
+                                    "type": [
+                                        "object",
+                                        "string"
+                                    ]
+                                }
+                            }
+                        },
                         "annotations": {
                             "type": [
                                 "object",
diff -up vault/values.yaml.0.26.1 vault/values.yaml
--- vault/values.yaml.0.26.1	2023-11-07 14:13:00.865329166 +0100
+++ vault/values.yaml	2023-11-07 14:14:15.318633813 +0100
@@ -673,6 +673,19 @@ server:
       # YAML-formatted multi-line templated string map of the annotations to apply
       # to the standby service.
       annotations: {}
+
+    nonha:
+      # Extra annotations for the service definition. This can either be YAML or a
+      # YAML-formatted multi-line templated string map of the annotations to apply
+      # to the service.
+      annotations: {}
+
+    internal:
+      # Extra annotations for the service definition. This can either be YAML or a
+      # YAML-formatted multi-line templated string map of the annotations to apply
+      # to the service.
+      annotations: {}
+
     # If enabled, the service selectors will include `app.kubernetes.io/instance: {{ .Release.Name }}`
     # When disabled, services may select Vault pods not deployed from the chart.
     # Does not affect the headless vault-internal service with `ClusterIP: None`
